### PR TITLE
map_afrom_cp: unescape double-quoted and back-quoted RHS prior to parsing it

### PR DESCRIPTION
**Pull request by mkhon**
@alandekok 

This is an updated patch from #4590. 
All test cases now pass.

### CODE REVIEW

**Comment by mkhon** [comment]
@alandekok @arr2036 can you please review this one?

**Comment by mkhon** [comment]
@alandekok can this one get reviewed, pleaae?

File: src/lib/server/map.c [hunk 1]
```diff
@@ -164,11 +165,25 @@ int map_afrom_cp(TALLOC_CTX *ctx, map_t **out, map_t *parent, CONF_PAIR *cp,
 	 *	RHS might be an attribute reference.
 	 */
 	type = cf_pair_value_quote(cp);
+	p_rules = value_parse_rules_unquoted[type]; /* We're not searching for quotes */
+	if (type == T_DOUBLE_QUOTED_STRING || type == T_BACK_QUOTED_STRING) {
+		char *unescaped_value;
+		slen = fr_sbuff_out_aunescape_until(map, &unescaped_value,
+				&FR_SBUFF_IN(value, strlen(value)), SIZE_MAX, p_rules->terminals, p_rules->escapes);
```
Comment (alandekok) [inline src/lib/server/map.c:172]: `strlen(value)` could be `talloc_array_length(value) - 1`, which avoids looping over the entire value

File: src/lib/server/map.c [hunk 2]
```diff
@@ -164,11 +165,25 @@ int map_afrom_cp(TALLOC_CTX *ctx, map_t **out, map_t *parent, CONF_PAIR *cp,
...
+		slen = strlen(value);
+	}
 
 	slen = tmpl_afrom_substr(map, &map->rhs,
-				 &FR_SBUFF_IN(value, strlen(value)),
+				 &FR_SBUFF_IN(value, slen),
 				 type,
-				 value_parse_rules_unquoted[type],	/* We're not searching for quotes */
+				 p_rules,
 				 rhs_rules);
```
Comment (alandekok) [inline src/lib/server/map.c:187]: `unescaped_value` should be freed after the call to `tmpl_afrom_substr()`.  The value isn't used any more, and there's no reason to keep it around.

File: src/lib/server/map.c (hunk 1 above)
Comment (mkhon) [reply src/lib/server/map.c]: Done

File: src/lib/server/map.c (hunk 2 above)
Comment (mkhon) [reply src/lib/server/map.c:188]: Done

**Comment by mkhon** [comment]
@alandekok can you please review an updated PR?

**Comment by mkhon** [comment]
@alandekok can you please review the updated PR?

**Comment by alandekok** [comment]
I've merged it in, and I'll push some additional fixes.

**Pull request merged**